pyvrp/cpp/search/Exchange.h

```diff
@@ -147,13 +152,24 @@ Cost Exchange<N, M>::evalRelocateMove(Route::Node *U,
     }
     else  // within same route
     {
+        deltaCost
+            -= costEvaluator.loadPenalty(uRoute->load(), uRoute->capacity());
         deltaCost -= costEvaluator.twPenalty(uRoute->timeWarp());
 
         if (deltaCost >= 0)
             return deltaCost;
 
         if (U->idx() < V->idx())
         {
+            auto const ls = LoadSegment::merge(
+                uRoute->before(U->idx() - 1),
+                uRoute->between(U->idx() + N, V->idx()),
+                uRoute->between(U->idx(), U->idx() + N - 1),
+                uRoute->after(V->idx() + 1));
+
+            deltaCost
+                += costEvaluator.loadPenalty(ls.load(), uRoute->capacity());
+
```

Comment:
All this duplication with essentially the same arguments to `TimeWindowSegment` just below is somewhat ugly. As a next step after this PR, we might consider implementing a concatenation scheme for distances as well, and then a generic `cost` (?) function that evaluates a given new route order - i.e., forwards all those arguments to DistanceSegment, LoadSegment, and TimeWindowSegment to obtain a cost.

Or something similar. I haven't completely figured out the interface for such a function.